docs/knn/index.md

```diff
@@ -246,3 +246,98 @@ All parameters are required.
 The standard KNN query and custom scoring option perform differently. Test using a representative set of documents to see if the search results and latencies match your expectations.
 
 Custom scoring works best if the initial filter reduces the number of documents to no more than 20,000. Increasing shard count can improve latencies, but be sure to keep shard size within [the recommended guidelines](../elasticsearch/#primary-and-replica-shards).
+
+
+## Use Similarity methods in painless scripting
```

Comment:
Use similarity functions in Painless scripts

From the Painless docs, it looks like they are functions rather than methods. We just want to choose the right term and be consistent.